src/Conf.ml

```diff
@@ -360,172 +486,90 @@ let validate () =
-          ( match value with
-          | "compact" -> `Compact
-          | "sparse" -> `Sparse
-          | other ->
-              user_error
-                (Printf.sprintf "Unknown type-decl value: %S" other)
-                [] ) }
-  | "escape-chars" ->
-      { conf with
-        escape_chars=
-          ( match value with
-          | "decimal" -> `Decimal
-          | "hexadecimal" -> `Hexadecimal
-          | "preserve" -> `Preserve
-          | other ->
-              user_error
-                (Printf.sprintf "Unknown escape-chars value: %S" other)
-                [] ) }
-  | "escape-strings" ->
-      { conf with
-        escape_strings=
-          ( match value with
-          | "decimal" -> `Decimal
-          | "hexadecimal" -> `Hexadecimal
-          | "preserve" -> `Preserve
-          | other ->
-              user_error
-                (Printf.sprintf "Unknown escape-strings value: %S" other)
-                [] ) }
-  | "break-string-literals" ->
-      { conf with
-        break_string_literals=
-          ( match value with
-          | "never" -> `Never
-          | "newlines" -> `Newlines
-          | "wrap" -> `Wrap
-          | other ->
-              user_error
-                (Printf.sprintf "Unknown break-string-literals value: %S"
-                   other)
-                [] ) }
-  | "parens-tuple" ->
-      { conf with
-        parens_tuple=
-          ( match value with
-          | "always" -> `Always
-          | "multi-line-only" -> `Multi_line_only
-          | other ->
-              user_error
-                (Printf.sprintf "Unknown parens-tuple value: %S" other)
-                [] ) }
   | "version" when not !no_version_check ->
-      if String.equal Version.version value then conf
-      else
-        user_error
-          ( "version mismatch: .ocamlformat requested " ^ value
-          ^ " but version is " ^ Version.version )
-          []
-  | "wrap-comments" -> {conf with wrap_comments= Bool.of_string value}
-  | "ocp-indent-compat" ->
-      {conf with ocp_indent_compat= Bool.of_string value}
-  | _ -> conf
-
-let rec read_conf_files conf dir =
+      if String.equal Version.version value then Ok config
+      else Error (`Bad_value (value, name))
+  | name -> C.update_config ~config ~name ~value ~inline
+
+let parse_config config ~inline s =
+  let s =
+    match String.index s '#' with
+    | Some i -> String.sub s ~pos:0 ~len:i
+    | None -> s
+  in
+  let s = String.strip s in
+  match String.split ~on:'=' s with
+  | [] | [""] -> Ok config
+  | [var; value] ->
+      update ~config ~name:(String.strip var) ~value:(String.strip value)
+        ~inline
+  | [s] -> (
+    match String.split ~on:' ' s with
+    | [] | [""] -> impossible "previous match"
+    | [var; value] ->
+        (* tolerate space separated [var value] to compatibility with older
+           config file format *)
+        update ~config ~name:(String.strip var) ~value:(String.strip value)
+          ~inline
```

Comment:
I'm tempted to issue a deprecation warning to stderr in this case, and remove this fallback in the next version.

Reply:
done